addons/analytic/tests/common.py

```diff
@@ -41,3 +42,15 @@ def setup_independent_company(cls, **kwargs):
     def get_default_groups(cls):
         groups = super().get_default_groups()
         return groups + cls.env.ref('analytic.group_analytic_accounting')
+
+    @classmethod
+    def setup_independent_user(cls):
+        return new_test_user(
+            cls.env,
+            name='The anal(ytic) expert!',
+            login='analytic',
+            password='analytic',
+            email='analyticman@test.com',
+            groups_id=cls.env.user.groups_id.ids + [cls.env.ref('analytic.group_analytic_accounting').id],
```

Comment:
Ideally, this should use `get_default_groups` so we would always use these (and not depend on whether we would enter this one or an other based on the order of installation). 
It might not be necessary now, as there extremely few uses of this method. But it could be more important in 18.2/3 when we change the groups on the user.

Tho, you would need to add `group_system` to be able to create companies (ideally you would just add sudos when needed and so, in the base common for instance).